Tools/WebKitTestRunner/cocoa/UIScriptControllerCocoa.mm

```diff
@@ -336,4 +336,16 @@ - (void)paste:(id)sender;
     }];
 }
 
+void UIScriptControllerCocoa::requestRenderedTextForSelector(JSStringRef selectorRef, JSValueRef callback)
+{
+    unsigned callbackID = m_context->prepareForAsyncTask(callback, CallbackTypeNonPersistent);
+    [webView() _requestRenderedTextForElementSelector:toWTFString(selectorRef) completionHandler:^(NSString *text, NSError *error) {
+        if (!m_context)
+            return;
+
+        auto result = adopt(JSStringCreateWithCFString((__bridge CFStringRef)(error.description ?: text)));
```

Comment:
I wonder if we should use `JSRetainPtr` rather than `auto` due to the explicit smart pointer usage guidelines.